src/Razor/test/Microsoft.VisualStudio.Razor.Integration.Test/AbstractIntegrationTest.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Threading;
+using Microsoft.VisualStudio.Razor.Integration.Test.InProcess;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using Xunit;
+using Xunit.Sdk;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.Razor.Integration.Test
+{
+    // TODO: Start collecting LogFiles on failure
+
+    /// <remarks>
+    /// The following is the xunit execution order:
+    ///
+    /// <list type="number">
+    /// <item><description>Instance constructor</description></item>
+    /// <item><description><see cref="IAsyncLifetime.InitializeAsync"/></description></item>
+    /// <item><description><see cref="BeforeAfterTestAttribute.Before"/></description></item>
+    /// <item><description>Test method</description></item>
+    /// <item><description><see cref="BeforeAfterTestAttribute.After"/></description></item>
+    /// <item><description><see cref="IAsyncLifetime.DisposeAsync"/></description></item>
+    /// <item><description><see cref="IDisposable.Dispose"/></description></item>
+    /// </list>
+    /// </remarks>
+    [IdeSettings(MinVersion = VisualStudioVersion.VS2022, RootSuffix = "RoslynDev")]
+    public abstract class AbstractIntegrationTest : IAsyncLifetime, IDisposable
+    {
+        protected const string ProjectName = "TestProj";
+        protected const string SolutionName = "TestSolution";
+
+        /// <summary>
+        /// A long timeout used to avoid hangs in tests, where a test failure manifests as an operation never occurring.
+        /// </summary>
+        public static readonly TimeSpan HangMitigatingTimeout = TimeSpan.FromMinutes(4);
```

Comment:
Hard to say since it hasn't passed yet due to the Preview1 issue. When I run them locally it's a couple minutes, with most of that time being spent starting/setting up the VS instance, which gets re-used (unless otherwise requested by the test). Therefore additional tests shouldn't add that much time on top (unless they fail by hitting this timeout).

Reply:
Surprisingly the answer seems to be ~12-15 minutes ([based on the log](https://dev.azure.com/dnceng/_apis/resources/Containers/9160174/Windows_NT_Windows%20debug%20TestResults?itemPath=Windows_NT_Windows%20debug%20TestResults%2FMicrosoft.VisualStudio.Razor.Integration.Test_net472_x64.html) from the most recent run.

@sharwell do you happen to know what might be causing that time? The actual runs are listed as ~9 seconds, so I assume the time is spent in some kind of VS setup that can hopefully be avoided or parallelized?

Reply:
So, after some of the recent fixes I can now see that while the Integration tests take ~8 minutes to run they don't actually extend the build time by that much (if at all). My successful build took ~32 minutes, other PR builds without my changes take 23-32 minutes.